modules/virt-defining-a-watchdog.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/advanced_vm_management/virt-configuring-a-watchdog.adoc
+
+[id="virt-defining-a-watchdog"]
+= Defining a watchdog device
+
+.Procedure
+
+. Choose the watchdog action. This defines how the watchdog proceeds when the operating system (OS) no longer responds.
+
++
+.Available actions
+[horizontal]
+`poweroff`:: The virtual machine (VM) powers down, and the guest OS cannot react.
+`reset`:: The VM reboots in place, and the guest OS cannot react. Because the length of time required for the guest OS to reboot can cause liveness probes to timeout, you should avoid their use.
+`shutdown`:: The VM powers off gracefully and becomes non-runnable.
+
+. Create a YAML file and apply it to the cluster.
++
+The example below configures the i6300esb watchdog device on a Fedora VM with the poweroff action and exposes the device as `/dev/watchdog`.
++
+[source,yaml]
+----
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-with-watchdog
+  name: vmi-with-watchdog
+spec:
+  domain:
+    devices:
+      watchdog:
+        name: mywatchdog
+        i6300esb:
+          action: "poweroff"
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 512M
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - containerDisk:
+      image: kubevirt/fedora-cloud-container-disk-demo
+    name: containerdisk
+----
+
+This device can now be used by the watchdog binary. The following example will send a heartbeat every two seconds to `/dev/watchdog` and execute the defined action after four seconds without a heartbeat. This command must be run inside the guest.
+
+[source,terminal]
+----
+$ watchdog -t 2000ms -T 4000ms /dev/watchdog
+----
+
+.Verification
+. Run the following command on the virtual machine to confirm the VM identified the watchdog card:
+
+[source,terminal]
+----
+$ lspci | grep watchdog -i
+----
+
+. Run one of the following commands to confirm the watchdog is active:
```

Comment:
Was curious to know, how the doc existed for RHV and came across the below:
https://access.redhat.com/documentation/en-us/red_hat_virtualization/4.4/html/virtual_machine_management_guide/sect-configuring_a_watchdog

Feel it would be good to have a warning before this step, similar to what we have for in RHV.

Reply:
Is this the warning you want to mirror?

WARNING
This procedure is provided for testing the functionality of watchdogs only and must not be run on production machines.

Reply:
Yes.